src/els_client.erl

```diff
@@ -47,25 +49,23 @@
 %% Defines
 %%==============================================================================
 -define(SERVER, ?MODULE).
--define(JSON_OPTS, [return_maps, {labels, atom}]).
 
 %%==============================================================================
 %% Record Definitions
 %%==============================================================================
--record(state, { transport :: tcp | stdio
-               , connection
-               , request_id = 1
-               , pending    = []
-               , buffer   = <<>>
+-record(state, { transport_cb     :: transport_cb()
+               , transport_server :: pid()
```

Comment:
This may change in the future, since the STDIO implementation is abusing this field to store the reference to the IODEVICE.

Reply:
I'm not sure what this comment means 😄

Reply:
Yeah, not the clearest one. What I mean is that the field is named "transport_server" and it is supposed to store a Pid, but in the case of the stdio implementation we store the iodevice in there. And Dialyzer doesn't even seem to be able to figure out the discrepancy.

Reply:
`iodevice()`s are `pid()`s AFAIK.